src/nvim/hashtab.h

```diff
@@ -81,10 +81,10 @@ typedef struct hashtable_S {
       size_t hi##todo_ = hi##ht_->ht_used; \
       for (hashitem_T *hi = hi##ht_->ht_array; hi##todo_; hi++) { \
         if (!HASHITEM_EMPTY(hi)) { \
+          hi##todo_--; \
```

Comment:
This change is required as any code inside `code` that has a `continue` statement will cause us to skip decrementing `hi##todo_`, causing an oob when we read more of the hash table than those that are in use. This feels like the simplest fix, rather than requiring that any code within `HASHTAB_ITER` (or any macros that use it) avoid the use of `continue`.